### ISSUE TITLE
STL file issue

### ISSUE THREAD

**Issue by CEBYMS** [tpms-lattice/ASLI#1]
The .STL file generated using Solidworks is not getting loaded in ASLI. Please comment on the procedure and specifics for the .STL file generation.

The file name was assigned as Part1. Even though, it says Part1.STL.stl in the error message.
The error is as follows:

SETTING UP PROBLEM...

  Opening config.yml

  Error:  Unable to open file H:/ASLI_v0.1-Windows/inputs/Part1.STL.stl

**Comment by tpms-lattice** [comment]
Can you share the file so that I can have a look?

**Comment by CEBYMS** [comment]
Yes, Sure. I'm not able to upload the .stl file, here. Can you please share your email id?

**Comment by tpms-lattice** [comment]
Got your file and I have noticed two different issues, both seem to be related to the TetGen library used to parse the input files in ASLI. The first one is that it looks like TetGen is expecting the extension to be lower case and adds it when it is not. This leads to an incorrect filename and thus the “Unable to open file” error. As a workaround I would suggest to change the extension of your file to lower case manually.
The second issue I encountered was upon changing the extension to lower case, i.e. I got the error “Wrong number of vertices in file inputs/Part1.stl”. Upon looking into it, it seems to be an issue with the binary stl parser of TetGen. A similar problem has already been pointed out by someone some years ago (see https://github.com/inducer/meshpy/issues/21) but it seems it has still not been fixed. Opening the .stl file in another program like MeshLab and saving it again fixed the problem for me.
I will look into fixing both issues, but at the moment I am quite busy so it may take a bit of time before I am able to do it. The workarounds should, however, get your around this issues for now.
